modules/system/area-helpers.js

```diff
@@ -1,25 +1,24 @@
 export default class AreaHelpers
 {
+    static SEMAPHORE = new foundry.utils.Semaphore(1);
     /**
      * Determines if a coordinate is within a Template's strokes
      *
      * @param {Object} {x, y} object being tested
      * @param {Template} template Template object being tested
      * @returns
      */
-    static isInTemplate(point, template)
-    {
-        if (template.document.t == "rect")
-        {
-            return this._isInRect(point, template);
+    static isInTemplate(token, template) {
+        let point = token.object.center;
+        //TODO: check if boneyard module is present and use it instead.
+        if (template.t == "rect") {
```

Comment:
yes, the goal was to handle various collision methods, and for that, we need object, probably it can be made more generic, to pass not only token but any placeable object or shape. Otherwise there would have to be another overwride method for point only.